capsules/src/net/icmpv6/icmpv6.rs

```diff
@@ -0,0 +1,182 @@
+//! ICMPv6 layer of the Tock networking stack.
+//!
+//! - Author: Conor McAvity <cmcavity@stanford.edu>
+
+use net::stream::{decode_u16, decode_u32, decode_u8};
+use net::stream::{encode_u16, encode_u32, encode_u8};
+use net::stream::SResult;
+
+#[derive(Copy, Clone)]
+pub struct ICMP6Header {
+    pub code: u8,
+    pub cksum: u16,
+    pub options: ICMP6HeaderOptions,
+    pub len: u16, // Not ICMP field, here for convenience
+}
+
+#[derive(Copy, Clone)]
+pub enum ICMP6HeaderOptions {
+    Type1 { unused: u32 },
+    Type3 { unused: u32 },
+    Type128 { id: u16, seqno: u16 },
+    Type129 { id: u16, seqno: u16 },
+}
+
+#[derive(Copy, Clone)]
+pub enum ICMP6Type {
+    Type1,   // Destination Unreachable
+    Type3,   // Time Exceeded
+    Type128, // Echo Request
+    Type129, // Echo Reply
+}
+
+impl ICMP6Header {
+    pub fn new(icmp_type: ICMP6Type) -> ICMP6Header {
```

Comment:
Maybe just have the caller pass in an `ICMP6HeaderOptions` (possibly renamed to `ICMP6Type`),  in which case the caller can actually _control_ the values for, e.g., `id` and `seqno`.